src/Compilers/Core/Portable/SpecialTypeExtensions.cs

```diff
@@ -142,6 +142,72 @@ public static bool IsValidEnumUnderlyingType(this SpecialType specialType)
             }
         }
 
+        public static bool IsNumericType(this SpecialType specialType)
+        {
+            switch (specialType)
+            {
+                case SpecialType.System_Byte:
+                case SpecialType.System_SByte:
+                case SpecialType.System_Int16:
+                case SpecialType.System_UInt16:
+                case SpecialType.System_Int32:
+                case SpecialType.System_UInt32:
+                case SpecialType.System_Int64:
+                case SpecialType.System_UInt64:
+                case SpecialType.System_Single:
+                case SpecialType.System_Double:
+                case SpecialType.System_Decimal:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        public static bool IsUnsignedIntegralType(this SpecialType specialType)
+        {
+            switch (specialType)
+            {
+                case SpecialType.System_Byte:
+                case SpecialType.System_UInt16:
+                case SpecialType.System_UInt32:
+                case SpecialType.System_UInt64:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        public static bool IsSignedIntegralType(this SpecialType specialType)
+        {
+            switch (specialType)
+            {
+                case SpecialType.System_SByte:
+                case SpecialType.System_Int16:
+                case SpecialType.System_Int32:
+                case SpecialType.System_Int64:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        public static int VBForToShiftBits(this SpecialType specialType)
```

Comment:
Can we add a comment on the intent of this method or update the name. Unsure exactly what this is for at a glance.

Reply:
> Can we add a comment on the intent of this method or update the name. Unsure exactly what this is for at a glance. 

Will do

---
In reply to: [187118574](https://github.com/dotnet/roslyn/pull/26734#discussion_r187118574) [](ancestors = 187118574)